pkg/predicate/predicate.go

```diff
@@ -91,6 +94,26 @@ func (p Funcs) Generic(e event.GenericEvent) bool {
 	return true
 }
 
+// NewPredicateFuncs returns a predicate funcs that applies the given filter function
+// on CREATE, UPDATE, DELETE and GENERIC events. For UPDATE events, the filter is applied
+// to both the old and new object and OR's the result.
+func NewPredicateFuncs(filter func(meta metav1.Object, object runtime.Object) bool) Funcs {
+	return Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return filter(e.Meta, e.Object)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return filter(e.MetaOld, e.ObjectOld) || filter(e.MetaNew, e.ObjectNew)
```

Comment:
@jiachengxu IMHO it makes more sense to not OR but filter only on the new object

Reply:
Thanks for the review, I also think it makes more sense. Fixed.